packages/Core/tag.candy

```diff
@@ -0,0 +1,27 @@
+type = use "..type"
+[ifElse] = use "..controlFlow"
+
+is value := type.is value Tag
+
+getSymbol tag :=
+  needs (is tag)
+  tag | ✨.tagGetSymbol
+
+hasValue tag :=
+  needs (is tag)
+  tag | ✨.tagHasValue
+
+getValue tag :=
+  needs (is tag)
+  needs (hasValue tag)
+  tag | ✨.tagGetValue
+
+withSymbol tag symbolTag :=
+  needs (is tag)
+  needs (is symbolTag)
+  ifElse (tag | hasValue) { symbolTag (tag | getValue) } { symbolTag | getSymbol }
+
+withValue tag valueTag :=
+  needs (is tag)
+  needs (is valueTag)
+  ifElse (valueTag | hasValue) { tag (valueTag | getValue) } { tag | getSymbol }
```

Comment:
I would have expected this to take a tag and a value, for example, `tag.withValue Foo 5` (= `Foo 5`) or `tag.withValue (Foo 2) 3` (= `Foo 3`).